app/assets/javascripts/discourse/app/components/user-stream.gjs

```diff
@@ -0,0 +1,233 @@
+  @service dialog;
+  @service composer;
+  @service appEvents;
+  @service currentUser;
+  @tracked lastDecoratedElement;
+  @tracked filterClassName = "";
+
+  eventListeners = modifier((element) => {
+    $(element).on("click.details-disabled", "details.disabled", () => false);
+    $(element).on("click.discourse-redirect", ".excerpt a", (e) => {
+      return ClickTrack.trackClick(e, getOwner(this));
+    });
+    later(() => {
+      this.updateLastDecoratedElement();
+      this.appEvents.trigger("decorate-non-stream-cooked-element", element);
+    });
+
+    return () => {
+      $(element).off("click.details-disabled", "details.disabled");
+      // Unbind link tracking
+      $(element).off("click.discourse-redirect", ".excerpt a");
+    };
+  });
+
+  constructor() {
+    super(...arguments);
+    this.setFilterClassName();
+  }
+
+  setFilterClassName() {
+    const filter = this.args.stream?.filter;
+    if (filter) {
+      this.filterClassName = `filter-${filter.toString().replace(",", "-")}`;
+    }
+  }
+
+  @action
+  updateLastDecoratedElement() {
+    const nodes = document.querySelectorAll(".user-stream-item");
+    if (!nodes || nodes.length === 0) {
+      return;
+    }
+
+    const lastElement = nodes[nodes.length - 1];
+    if (lastElement === this.lastDecoratedElement) {
+      return;
+    }
+    this.lastDecoratedElement = lastElement;
+  }
+
+  @action
+  async removeBookmark(userAction) {
+    try {
+      await Post.updateBookmark(userAction.get("post_id"), false);
+      this.args.stream.remove(userAction);
+    } catch (error) {
+      popupAjaxError(error);
+    }
+  }
+
+  @action
+  async resumeDraft(item) {
+    if (this.composer.get("model.viewOpen")) {
+      this.composer.close();
+    }
+    if (item.get("postUrl")) {
+      DiscourseURL.routeTo(item.get("postUrl"));
+    } else {
+      try {
+        const draftData = await Draft.get(item.draft_key);
+        const draft = draftData.draft || item.data;
+        if (!draft) {
+          return;
+        }
+        this.composer.open({
+          draft,
+          draftKey: item.draft_key,
+          draftSequence: draftData.draft_sequence,
+        });
+      } catch (error) {
+        popupAjaxError(error);
+      }
+    }
+  }
+
+  @action
+  removeDraft(draft) {
+    this.dialog.yesNoConfirm({
+      message: i18n("drafts.remove_confirmation"),
+      didConfirm: async () => {
+        try {
+          await Draft.clear(draft.draft_key, draft.sequence);
+          this.args.stream.remove(draft);
+
+          if (draft.draft_key === NEW_TOPIC_KEY) {
+            this.currentUser.has_topic_draft = false;
+          }
+        } catch (error) {
+          popupAjaxError(error);
+        }
+      },
+    });
+  }
+
+  @action
+  async loadMore() {
+    await this.args.stream.findItems();
+
+    if (this.args.stream.canLoadMore === false) {
+      return [];
+    }
+
+    later(() => {
+      let element = this.lastDecoratedElement?.nextElementSibling;
```

Comment:
Could we make this a proper modifier on the elements we want to operate on? It's a bit convoluted to use `later()` and then iterate through the DOM 🤔